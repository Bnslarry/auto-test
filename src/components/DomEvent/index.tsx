import { FC } from 'react';

interface IProps {
  onClick: () => void;
}

export const DomEvent: FC<IProps> = ({ onClick }) => {
  return (
    <div role="note" onClick={onClick}>
      点我试试
    </div>
  );
};
