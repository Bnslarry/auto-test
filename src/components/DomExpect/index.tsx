import { FC } from 'react';
import { Form } from '@douyinfe/semi-ui';

interface IProps {}

export const DomExpect: FC<IProps> = () => {
  return (
    <div>
      <div aria-label="empty_note"></div>
      <div
        role="note"
        style={{ display: 'none' }}
        className="test hidden"
        aria-hidden
      >
        1234
      </div>
      <div role="note">8459</div>
      {/* 表单验证 */}
      <form aria-label="form">
        <input
          type="text"
          name="username"
          disabled
          aria-disabled
          defaultValue="glow"
          aria-label="form_username"
        />
        <input
          type="number"
          name="age"
          defaultValue={26}
          required
          aria-label="form_age"
        />
        <input
          type="radio"
          name="sex"
          value="man"
          defaultChecked
          aria-checked
          aria-label="form_sex"
        />
        <input type="radio" name="sex" value="man" aria-label="form_sex" />
      </form>
      {/* semi 表单验证 */}
      <Form
        initValues={{ username: 'glow', age: 26, sex: 'man', hobby: 'code' }}
        aria-label="semi-form"
      >
        <Form.Input field="username" disabled name="username" />
        <Form.InputNumber field="age" required name="age" />
        <Form.RadioGroup field="sex" name="sex">
          <Form.Radio value="man" />
          <Form.Radio value="woman" />
        </Form.RadioGroup>
        <Form.Select field="hobby" name="hobby">
          <Form.Select.Option value="code">code</Form.Select.Option>
          <Form.Select.Option value="read">read</Form.Select.Option>
        </Form.Select>
      </Form>
    </div>
  );
};
