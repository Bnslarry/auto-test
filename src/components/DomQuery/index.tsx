import { FC } from "react";

interface IProps {}

export const DomQuery: FC<IProps> = () => {
  return (
    <div>
      <div>test1</div>
      <div>test2</div>
      <button aria-describedby="description">
          <div id="description">自定义aria按钮</div>
      </button>
      <div aria-label="test_note">1234</div>
    </div>
  );
};