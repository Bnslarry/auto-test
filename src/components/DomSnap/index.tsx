import { FC } from 'react';
interface IProps {}

export const DomSnap: FC<IProps> = () => {
  return (
    <form aria-label="form">
      <input
        type="text"
        name="username"
        disabled
        aria-disabled
        defaultValue="glow"
        aria-label="form_username"
      />
      <input
        type="number"
        name="age"
        defaultValue={26}
        required
        aria-label="form_age"
      />
      <input
        type="radio"
        name="sex"
        value="man"
        defaultChecked
        aria-checked
        aria-label="form_sex"
      />
      <input type="radio" name="sex" value="man" aria-label="form_sex" />
    </form>
  );
};
