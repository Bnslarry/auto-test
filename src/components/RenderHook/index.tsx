import { FC } from 'react';
import useCount from './useCount';

interface IProps {}

export const RenderHook: FC<IProps> = () => {
  const { num, increase } = useCount();

  return (
    <div>
      <span role="note">{num}</span>
      <button onClick={increase}>增加</button>
    </div>
  );
};
