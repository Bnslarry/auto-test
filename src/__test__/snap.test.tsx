import { render, screen } from '@testing-library/react';
import { DomSnap } from '../components/DomSnap';

describe('examples for snap', () => {
  test('a test for component snap', () => {
    const { baseElement } = render(<DomSnap />);
    expect(baseElement).toMatchSnapshot();
  });

  test('a test for part component snap', () => {
    render(<DomSnap />);
    expect(
      screen.getByRole('textbox', { name: 'form_username' })
    ).toMatchSnapshot();
  });
  test('a test for string snap', () => {
    expect('a test for string snap').toMatchSnapshot();
  });
});
