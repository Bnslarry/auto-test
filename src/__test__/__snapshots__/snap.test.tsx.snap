// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples for snap a test for component snap 1`] = `
<body>
  <div>
    <form
      aria-label="form"
    >
      <input
        aria-disabled="true"
        aria-label="form_username"
        disabled=""
        name="username"
        type="text"
        value="glow"
      />
      <input
        aria-label="form_age"
        name="age"
        required=""
        type="number"
        value="26"
      />
      <input
        aria-checked="true"
        aria-label="form_sex"
        checked=""
        name="sex"
        type="radio"
        value="man"
      />
      <input
        aria-label="form_sex"
        name="sex"
        type="radio"
        value="man"
      />
    </form>
  </div>
</body>
`;

exports[`examples for snap a test for part component snap 1`] = `
<input
  aria-disabled="true"
  aria-label="form_username"
  disabled=""
  name="username"
  type="text"
  value="glow"
/>
`;

exports[`examples for snap a test for string snap 1`] = `"a test for string snap"`;
